pageserver/src/http/routes.rs

```diff
@@ -1521,6 +1524,71 @@ async fn timeline_gc_unblocking_handler(
     block_or_unblock_gc(request, false).await
 }
 
+/// Traces GetPage@LSN requests for a timeline, and emits metadata in an efficient binary encoding.
+/// Use the `pagectl page-trace` command to decode and analyze the output.
+async fn timeline_page_trace_handler(
+    request: Request<Body>,
+    cancel: CancellationToken,
+) -> Result<Response<Body>, ApiError> {
+    let tenant_shard_id: TenantShardId = parse_request_param(&request, "tenant_shard_id")?;
+    let timeline_id: TimelineId = parse_request_param(&request, "timeline_id")?;
+    let state = get_state(&request);
+    check_permission(&request, Some(tenant_shard_id.tenant_id))?;
```

Comment:
Passing `Some(tenant_id)` here is wrong, it will allow the compute to catpure a page trace.

Reply:
Yeah, I saw some other seemingly-internal endpoints pass the tenant ID as well. Didn't seem like a problem in and of itself that the tenant can see this data, but there's no good reason why they should either. Fixed.